test/spec/Feature/Query/RelatedQueriesSpec.hs

```diff
@@ -258,8 +258,10 @@ spec = describe "related queries" $ do
         , matchHeaders = [matchContentTypeJson]
         }
 
-    -- "?table=not.is.null" does a "table IS DISTINCT FROM NULL" instead of a "table IS NOT NULL"
-    -- https://github.com/PostgREST/postgrest/issues/2800#issuecomment-1720315818
+    -- "?table=not.is.null" translates to:
+    -- * "table.join_column IS NOT NULL" for to-one relationships
+    -- * "table IS NOT NULL" for to-many relationships
```

Comment:
The query for "to-many" rels forces me to use a `GROUP BY`, the original related issue also happened for these cases only. If no better query exists then it will be left as `table IS NOT NULL` or `table IS DISTINCT FROM NULL`.